packages/api-derive/src/bounties/bounties.ts

```diff
@@ -5,20 +5,24 @@ import type { ApiInterfaceRx } from '@polkadot/api/types';
 import type { DeriveBounties } from '@polkadot/api-derive/types';
 import type { Bytes, Option } from '@polkadot/types';
 import type { Bounty, BountyIndex } from '@polkadot/types/interfaces';
-import type { Observable } from '@polkadot/x-rxjs';
 
 import { memo } from '@polkadot/api-derive/util';
-import { combineLatest } from '@polkadot/x-rxjs';
+import { Codec } from '@polkadot/types/types';
+import { combineLatest, Observable, of } from '@polkadot/x-rxjs';
 import { map, switchMap } from '@polkadot/x-rxjs/operators';
 
-type Result = [Option<Bounty>[], Option<Bytes>[]];
+type Result = [Option<Bounty>[], Option<Bytes>[], Codec[]];
 
-function parseResult ([maybeBounties, maybeDescriptions]: Result): DeriveBounties {
+function parseResult (api: ApiInterfaceRx, [maybeBounties, maybeDescriptions, ids]: Result): DeriveBounties {
   const bounties: DeriveBounties = [];
 
   maybeBounties.forEach((bounty, index) => {
     if (bounty.isSome) {
-      bounties.push({ bounty: bounty.unwrap(), description: maybeDescriptions[index].unwrapOrDefault().toUtf8() });
+      bounties.push({
+        bounty: bounty.unwrap(),
+        description: maybeDescriptions[index].unwrapOrDefault().toUtf8(),
+        index: api.registry.createType('BountyIndex', ids[index])
```

Comment:
This will already be a BountyIndex as per line 41

Reply:
ok, improved 🙂